### PR TITLE
Add interop docs and optimizers

### PR DESCRIPTION
**Pull request by dstrain115**
- Move QASM from circuits doc to a new interop page
- Add JSON and quirk functionality to interop page
- Add a section on optimizers to circuits page.

### CODE REVIEW

**Comment by dstrain115** [comment]
Ping.  Would it be possible to get a review on this PR?

File: docs/circuits.ipynb [hunk 1]
```diff
@@ -41,14 +41,28 @@
...
+    },
     {
       "cell_type": "markdown",
       "metadata": {
         "id": "77zB_UqkAwCR",
         "colab_type": "text"
       },
       "source": [
-        "![Circuits and Moments](https://cirq.readthedocs.io/en/latest/_static/CircuitMomentOperation.png)"
+        "![Circuits and Moments](https://cirq.readthedocs.io/en/stable/_images/CircuitMomentOperation.png)"
```
Comment (vtomole) [inline docs/circuits.ipynb:65]: Is this image not in `latest`? I thought we store all these things in `static`. Why the change in this link?

File: docs/circuits.ipynb [hunk 2]
```diff
@@ -773,63 +835,45 @@
...
-    },
-    {
-      "cell_type": "markdown",
-      "metadata": {
-        "id": "TXFjdV6BAwDP",
-        "colab_type": "text"
-      },
-      "source": [
-        "The following call will create a circuit defined by the input QASM string:"
+        "Optimizers will have a function `optimize_circuit()` that can be used to perform this optimization.  Here is a simple example that removes moments that are empty:"
```
Comment (vtomole) [inline docs/circuits.ipynb:846]: nit: change `moments that are empty` to `empty moments`

File: docs/circuits.ipynb (hunk 1 above)
Comment (dstrain115) [reply docs/circuits.ipynb:65]: Good catch.  I think this was a merge issue of some sorts, since I changed this in an early PR.
I will also go and double-check other changes to make sure I am not squashing anything once I update the branch.

File: docs/circuits.ipynb (hunk 2 above)
Comment (dstrain115) [reply docs/circuits.ipynb:846]: Done.
